### ISSUE TITLE
Feature request: Partial upload/deploy

### ISSUE THREAD

**Issue by joshlf** [actions/deploy-pages#349]
TLDR: Can this Action support [this workflow](https://stackoverflow.com/questions/78460845/edit-part-of-github-pages-via-action-without-overwriting) natively?

My project currently uses [this configuration](https://github.com/google/zerocopy/blob/e86e363e4f2f06d83c71519c90b6e9950ede7184/.github/workflows/docs.yml#L102-L116) to publish to GitHub Pages from the `main` branch:

```yaml
      - name: Upload Cargo doc output to GitHub Pages
        uses: actions/upload-pages-artifact@v3
        with:
          path: target/doc
  deploy:
    name: Deploy to GitHub Pages
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
```

This generates rustdoc and then uploads the generated files to GitHub Pages.

I'm hoping to extend our GitHub pages to also upload a copy of the generated rustdoc for each in-flight PR so that we can preview what the docs will look like. I believe this should be possible with only a small addition to this Action: namely, the ability to specify a *target subdirectory* when deploying an artifact. This should in theory only require one extra input to the Action:

```yaml
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
        target: a/b/c
```

The `target` input would be optional, and default to `.`. It would instruct the Action to overwrite not necessarily the entire deployment, but a specific subdirectory of the deployment. This would allow us to write something like:

```yaml
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v4
        target: pulls/${{ ... }}
```

...and invoke this from a workflow triggered by each PR.

**Comment by sergio-bobillier** [comment]
I have a similar use case just with [YARD](https://rubydoc.info/gems/yard/file/docs/GettingStarted.md) documentation and I would also like to upload an HTML Tests and Coverage report.

Right now I'm just re-running the tests in my "Generate Documentation" action but I'd rather upload these things from my normal "RSpec Tests" action (on merge).

**Comment by imercerwillow** [comment]
One workaround is to use the cache. At the start of the workflow  run, retrieve the merged website with a folder for each component. Wipe the appropriate subfolder and copy the results into it. Now publish the whole lot to github pages and save the cache back ready for the next workflow.

**Comment by Glavin001** [comment]
Has anyone found a good workaround and could provide an example?

I was able to use https://github.com/peaceiris/actions-gh-pages however it uses `gh-pages` instead of actions, which bloats my repo size signifiacntly.

e.g.
```yaml
        uses: peaceiris/actions-gh-pages@v4
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./report
          destination_dir: ${{ steps.context.outputs.deployment_path }}
          keep_files: true
```

**Comment by dkaszews** [comment]
This would be really useful for hosting documentation, so that you can easily have both latest stable and dev versions available (plus aforementioned preview from PRs)
